File: backend/bang-ggood/src/main/java/com/bang_ggood/article/domain/Article.java
package com.bang_ggood.article.domain;

import com.bang_ggood.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import java.util.Objects;

import static lombok.AccessLevel.PROTECTED;

@Getter
@NoArgsConstructor(access = PROTECTED)
@Entity
public class Article extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String content;

    private String keyword;

    private String summary;

    private String thumbnail;

    public Article(String title, String content, String keyword, String summary, String thumbnail) {
        this.title = title;
        this.content = content;
        this.keyword = keyword;
        this.summary = summary;
        this.thumbnail = thumbnail;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Article article = (Article) o;
        return Objects.equals(id, article.id);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }

    @Override
    public String toString() {
        return "Article{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", content='" + content + '\'' +
                ", keyword='" + keyword + '\'' +
                ", summary='" + summary + '\'' +
                ", thumbnail='" + thumbnail + '\'' +
                '}';
    }
}

File: frontend/src/types/article.ts
export type ArticleType =
  | '계약 꿀팁'
  | '자취 일기'
  | '생활 꿀팁'
  | '동네 추천'
  | '방끗 활용법'
  | '우테코 생활'
  | '자취 꿀팁';

export const ARTICLE_TYPES: ArticleType[] = [
  '계약 꿀팁',
  '자취 일기',
  '생활 꿀팁',
  '동네 추천',
  '방끗 활용법',
  '우테코 생활',
  '자취 꿀팁',
];

export interface Article {
  articleId: number;
  keyword: ArticleType;
  title: string;
  summary: string;
  content?: string;
  thumbnail?: string | null;
  createdAt: string;
}

export interface ArticlePostForm {
  keyword: string;
  title: string;
  summary: string;
  content: string;
  thumbnail?: string;
}
